red_queen/cli.py

```diff
@@ -0,0 +1,175 @@
+# ------------------------------------------------------------------------------
+# Part of Red Queen Project.  This file is distributed under the MIT License.
+# See accompanying file /LICENSE for details.
+# ------------------------------------------------------------------------------
+
+#!/usr/bin/env python3
+import os
+import platform
+import subprocess
+import configparser
+import click
+
+
+def benchmarkRetrieval():
+    benchmark_category = {}
+    benchmark_types = []
+    benchmarks = []
+    dir_path = "red_queen/games/"
+    for entry in os.scandir(dir_path):
+        if entry.is_dir():
+            benchmark_category[entry.name] = []
+            sub_dict = {}
+            for sub in os.scandir(f"{dir_path}{entry.name}"):
+                if not sub.name.startswith("_") and sub.name.endswith(".py") and sub.is_file():
+                    sub_dict[sub.name] = sub.path
+            benchmark_category[entry.name] = sub_dict
+
+    benchmark_types = list(benchmark_category.keys())
+    for benchmark_pairs in benchmark_category.values():
+        for keys in benchmark_pairs.keys():
+            benchmarks.append(keys)
+
+    return benchmark_category, benchmark_types, benchmarks
+
+
+def complier_retrieval():
+    complier_list = []
+    config = configparser.ConfigParser()
+    config.read("pytest.ini")
+    for complier in config["pytest"]["markers"].split("\n"):
+        if complier != "":
+            complier_list.append(complier)
+    # print(complier_list)
+    # This line tests to see if there is a complier specifed
+    return complier_list
+
+
+def result_retrieval():
+    results = {}
+    dir_path = "results"
+    for entry in os.scandir(dir_path):
+        if entry.is_file():
+            filename = entry.name
+            result_count = int(filename.split("_")[0])
+            results[result_count] = {entry.name: entry.path}
+    return results
+
+
+def run_benchmarks(pytest_paths: str, m_tag: str, compiler: str):
+    click.echo("benchmarks ran")
+    if platform.system() == "Windows":
+        subprocess.run(
+            [f"python -m pytest -s {pytest_paths} {m_tag}{compiler} --store"],
```

Comment:
To make this work on windows, the arguments list needs to be separated per argument. Consider using the `str.split()` method in order to separate your string argument into a list separated by whitespace. However, this only works well with Windows as I tested with Linux and it did not work.

```suggestion
            f"python -m pytest -s {pytest_paths} {m_tag}{compiler} --store".split(),
```